Freeze feature flags

mkangia opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->
Add a new tag for feature flags called "Frozen" and use that for feature flags that are meant to be replaced and should not be extended to new domains in the interim.

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
https://dimagi.atlassian.net/browse/SAAS-18979 & also updating two flags marked for deprecation to be frozen as mentioned [here](https://docs.google.com/spreadsheets/d/1kE56g6cy3CRz4XncoNNW4riXts3HYTDdOKqw4CmdeDg/edit?disco=AAABt9gD-7Q)
The new tag added is different from the `FrozenPrivilegeToggle` since that is used when there is a privilege added to GA the feature flag.
We have the deprecated tag for feature flags that are meant to be removed without any replacement.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
Multiple

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Just a tag change, should be safe. Migration is also safe.

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

mkangia committed:
freeze feature flags

mkangia committed:
add frozen tag to toggle edit permissions model tab_slug options

mkangia commented on corehq/toggles/migrations/0004_alter_toggleeditpermission_tag_slug.py:
```diff
@@ -0,0 +1,18 @@
+            name='tag_slug',
+            field=models.CharField(choices=[('solutions_open', 'solutions_open'), ('solutions_conditional', 'solutions_conditional'), ('solutions_limited', 'solutions_limited'), ('saas_conditional', 'saas_conditional'), ('solutions', 'solutions'), ('product', 'product'), ('custom', 'custom'), ('solutions_internal', 'solutions_internal'), ('release', 'release'), ('deprecated', 'deprecated'), ('fro ...
+        ),
+    ]
```
That is a good observation.

It's possible `0002` might fail if it has not been run already.
(I tried reverting migrations to `0001` and then run migrations again but that didn't give an error, which I am not sure of why)

Anyhow, since the migration `0002` was written more than 4 months ago, it's safe to assume that `0002` has been run on existing environments already.
And since the concerning operation in `0002` is skipped on fresh install, this should not be a problem for new environments.

mkangia replied:
Lets do it,
https://github.com/dimagi/commcare-hq/pull/37194/commits/8d19604330a112370087bdece0f15bc1fea4d168

gherceg reviewed (approved):
We generally try to split out code changes from migrations to make reverting easier if needed, but I understand this is a pretty small change so that is tedious in this case. Just ensuring you've thought through what would happen if you did need to revert, and what, if any, problems might come up because of that? Won't block on a response to that, but just wanted to flag.

mkangia committed:
use specific tags available at the time migration was written

avoid failure for tags added after the migration, with updated tag_slug field choices

mkangia committed:
update tags for two feature flags

mkangia commented:
Thanks for the review @gherceg 

Made some updates.
Also sneaked in a change requested in https://docs.google.com/spreadsheets/d/1kE56g6cy3CRz4XncoNNW4riXts3HYTDdOKqw4CmdeDg/edit?disco=AAABt9gD-7Q
